### PR TITLE
Audio loading status

### PR DESCRIPTION
**Pull request by hikari-no-yume**

### CODE REVIEW

File: hibikase/PlaybackWidget.cpp [hunk 1]
```diff
@@ -51,22 +50,46 @@ void PlaybackWidget::LoadAudio(std::unique_ptr<QIODevice> io_device)
 {
     if (m_worker)
     {
-        QMetaObject::invokeMethod(m_worker, "Stop");  // Ensure state gets set to stopped
+        disconnect(m_worker, &AudioOutputWorker::StateChanged, this, &PlaybackWidget::OnStateChanged);
+        disconnect(m_worker, &AudioOutputWorker::TimeUpdated, this, &PlaybackWidget::UpdateTime);
+        QMetaObject::invokeMethod(m_worker, "Stop");
```
Comment (JosJuice) [inline hibikase/PlaybackWidget.cpp]: The reason I added a call to Stop here was to ensure that OnStateChanged gets called with StoppedState. Is there a point in having this line after the disconnect calls?

File: hibikase/PlaybackWidget.h [hunk 2]
```diff
@@ -39,6 +39,7 @@ public slots:
     void Stop();
 
 signals:
+    void LoadResult(QString error);
```
Comment (JosJuice) [inline hibikase/PlaybackWidget.h]: This isn't how you typically would name a signal. I'd suggest a name like "Loaded" (or "AudioLoaded"?) instead.

File: hibikase/PlaybackWidget.cpp [hunk 3]
```diff
@@ -94,7 +117,8 @@ void PlaybackWidget::OnStateChanged(QAudio::State state)
 void PlaybackWidget::UpdateTime(std::chrono::milliseconds ms)
 {
     QString text;
-    if (ms.count() >= 0)
+    // If worker has been deleted, ignore time update
```
Comment (JosJuice) [inline hibikase/PlaybackWidget.cpp]: Could you explain in what situation UpdateTime would be called when m_worker has been deleted?

File: hibikase/PlaybackWidget.cpp (hunk 1 above)
Comment (hikari-no-yume) [reply hibikase/PlaybackWidget.cpp]: I don't want the stop event to overwrite the button text :p

File: hibikase/PlaybackWidget.cpp (hunk 3 above)
Comment (hikari-no-yume) [reply hibikase/PlaybackWidget.cpp]: It shouldn't be, but it happens for some reason. I guess queued events.

Comment (JosJuice) [reply hibikase/PlaybackWidget.cpp]: There is the call in OnStateChanged, I suppose, but that one should always have a count of -1 anyway...

Comment (JosJuice) [reply hibikase/PlaybackWidget.cpp]: I suppose it's fine, then. I'm just thinking about if it's weird to have a comment like this that applies to the first half of the condition but not the second half...

File: hibikase/PlaybackWidget.cpp (hunk 1 above)
Comment (JosJuice) [reply hibikase/PlaybackWidget.cpp]: Yes, but can't you remove this line entirely?

Comment (hikari-no-yume) [reply hibikase/PlaybackWidget.cpp]: In theory, but will the audio worker be destroyed quickly?

File: hibikase/PlaybackWidget.h (hunk 2 above)
Comment (hikari-no-yume) [reply hibikase/PlaybackWidget.h]: I want to convey that it could also return an error

File: hibikase/PlaybackWidget.cpp (hunk 1 above)
Comment (JosJuice) [reply hibikase/PlaybackWidget.cpp]: It will be destroyed as quickly as this call to Stop would happen.

File: hibikase/PlaybackWidget.h (hunk 2 above)
Comment (JosJuice) [reply hibikase/PlaybackWidget.h]: Hmm... Maybe LoadFinished or InitializationFinished? Does that sound less like it always succeeds?

**Pull request merged**